apps/dataset/views/document.py

```diff
@@ -238,6 +238,24 @@ def put(self, request: Request, dataset_id: str, document_id: str):
                     request.data
                 ))
 
+        class Batch(APIView):
+            authentication_classes = [TokenAuth]
+
+            @action(methods=['PUT'], detail=False)
+            @swagger_auto_schema(operation_summary="批量取消任务",
+                                 operation_id="批量取消任务",
+                                 request_body=DocumentApi.BatchCancel.get_request_body_api(),
+                                 manual_parameters=DocumentSerializers.Create.get_request_params_api(),
+                                 responses=result.get_default_response(),
+                                 tags=["知识库/文档"]
+                                 )
+            @has_permissions(
+                lambda r, k: Permission(group=Group.DATASET, operate=Operate.MANAGE,
+                                        dynamic_tag=k.get('dataset_id')))
+            def put(self, request: Request, dataset_id: str):
+                return result.success(
+                    DocumentSerializers.Batch(data={'dataset_id': dataset_id}).batch_cancel(request.data))
+
     class Refresh(APIView):
         authentication_classes = [TokenAuth]
 
```

Comment:
The provided code snippet contains several issues and areas for improvement:

1. **Duplicate Code**: The `Put` method within both `DocumentApi.put` and `DocumentApi.Batch` has duplicated logic. Consider combining these into a single base class to reduce redundancy.

2. **Authentication Classes Duplication**: Both views (`DocumenApi.Put`, `DocumentApi.Batch`) reference the same authentication class (`TokenAuth`). This duplication is unnecessary since they should all use the same authentication scheme.

3. **Manual Parameters Usage**: In `DocumentApi.Batch.put`, you're using `manual_parameters`, which is an internal feature of Swagger AutoSchema that allows setting parameters directly. However, it might be better to define them at the view's class level if they don't need to change dynamically.

4. **Operation Summary and ID Duplication**: In `DocumentApi.Batch.put`, the operation summary and ID are duplicated as "批量取消任务". Ensure consistency in your documentation.

5. **Response Default Value**: The default response value in `swagger_auto_schema.responses` doesn't seem relevant and shouldn't be hardcoded. Instead, consider retrieving this dynamically from a configuration or settings file.

6. **Tags Consistency**: While you've specified only one tag under different view classes, ensure that this matches any potential tagging conventions used elsewhere in your application.

7. **Permissions Lambda Logic**: The permission lambda function lacks clarity. It seems like the permissions check is based on specific keys ('group', 'operate') but not properly documented what `'k'` represents.

8. **Batch Cancel Method Calls**: Since `docSerializers.Batch.batch_cancel` expects a different input (`{'dataset_id': dataset_id}`) compared to `docserializers.Create.get_request_params_api()` (which appears to expect a list), these methods call may not work as expected unless there is additional preprocessing needed.

9. **View Detail Parameter Handling**: The current implementation assumes that `document_id` parameter is set when calling `Batch.cancel`. Be cautious about how this affects scenarios where documents do not exist.

To address these points, you could refactor the shared functionality by creating a base view class with common parameters and operations, then extend this base class for concrete actions like updating multiple items (`DocumentApi.Batch.put`). Additionally, review and potentially update your documentation to reflect changes made to the functions and their parameters.

These modifications will help improve code efficiency, readability, adhere to best practices, and ensure correct behavior across your API endpoints.